src/CoreText/CTFontDescriptor.cs

```diff
@@ -738,34 +739,67 @@ public CTFontDescriptor (CTFontDescriptorAttributes attributes)
 		[SupportedOSPlatform ("maccatalyst")]
 		[SupportedOSPlatform ("tvos")]
 #endif
-#if NET
 		[DllImport (Constants.CoreTextLibrary)]
-		[return: MarshalAs (UnmanagedType.I1)]
-		static unsafe extern bool CTFontDescriptorMatchFontDescriptorsWithProgressHandler (IntPtr descriptors, IntPtr mandatoryAttributes,
-			delegate* unmanaged<CTFontDescriptorMatchingState, IntPtr, bool> progressHandler);
+		static unsafe extern byte CTFontDescriptorMatchFontDescriptorsWithProgressHandler (IntPtr descriptors, IntPtr mandatoryAttributes, BlockLiteral* progressBlock);
+
+		public delegate bool CTFontDescriptorProgressHandler (CTFontDescriptorMatchingState state, CTFontDescriptorMatchingProgress progress);
+
+#if !NET
+		delegate byte ct_font_desctiptor_progress_handler_t (IntPtr block, CTFontDescriptorMatchingState state, IntPtr progress);
+		static ct_font_desctiptor_progress_handler_t static_MatchFontDescriptorsHandler = MatchFontDescriptorsHandler;
+
+		[MonoPInvokeCallback (typeof (ct_font_desctiptor_progress_handler_t))]
 #else
-		[DllImport (Constants.CoreTextLibrary)]
-		[return: MarshalAs (UnmanagedType.I1)]
-		static extern bool CTFontDescriptorMatchFontDescriptorsWithProgressHandler (IntPtr descriptors, IntPtr mandatoryAttributes,
-			Func<CTFontDescriptorMatchingState, IntPtr, bool> progressHandler);
+		[UnmanagedCallersOnly]
 #endif
+		static byte MatchFontDescriptorsHandler (IntPtr block, CTFontDescriptorMatchingState state, IntPtr progress)
+		{
+			var del = BlockLiteral.GetTarget<CTFontDescriptorProgressHandler> (block);
+			if (del is not null) {
+				var progressDictionary = Runtime.GetNSObject<NSDictionary> (progress)!;
+				var strongDictionary = new CTFontDescriptorMatchingProgress (progressDictionary);
+				var rv = del (state, strongDictionary);
+				return rv ? (byte) 1 : (byte) 0;
```

Comment:
We like to cast too much hehe
```suggestion
				return (byte) (rv ? 1 : 0);
```
Of course this is meant as a joke, no need to change a thing ;)